src/web/handlers.rs

```diff
@@ -376,3 +376,52 @@ pub fn put_bso(bso_req: BsoPutRequest) -> impl Future<Item = HttpResponse, Error
 pub fn get_configuration(creq: ConfigRequest) -> impl Future<Item = HttpResponse, Error = Error> {
     future::result(Ok(HttpResponse::Ok().json(creq.limits)))
 }
+
+/** Returns a status message indicating the state of the current server
+ *
+ * * version - Version information
+ * * state_data - Is the app state data present ("good" | "fail")
+ * * database - Database connection and status ("good" | "warn" | "fail")
+ * * status - Overall system status ("good" | "fail")
+ */
+pub fn heartbeat(hb: HeartbeatRequest) -> impl Future<Item = HttpResponse, Error = Error> {
+    use crate::server::ServerState;
+
+    let mut checklist = HashMap::new();
```

Comment:
just an FYI: [serde's json!](https://docs.serde.rs/serde_json/macro.json.html) usually makes this kind of thing easier. You get back a `Value` which *I think* can be passed to the final `json()` call (as it implements Serialize)